src/read.c

```diff
@@ -2365,10 +2369,14 @@ static avifBool avifParseTrackHeaderBox(avifTrack * track, const uint8_t * raw,
     track->width = width >> 16;
     track->height = height >> 16;
 
-    if ((track->width == 0) || (track->height == 0) || (track->width > (imageSizeLimit / track->height))) {
+    if ((track->width == 0) || (track->height == 0)) {
         avifDiagnosticsPrintf(diag, "Track ID [%u] has an invalid size [%ux%u]", track->id, track->width, track->height);
         return AVIF_FALSE;
     }
+    if (track->width > (imageSizeLimit / track->height)) {
+        avifDiagnosticsPrintf(diag, "Track ID [%u] size is too large [%ux%u]", track->id, track->width, track->height);
+        return AVIF_FALSE;
+    }
```

Comment:
Joe: I was planning to add the AVIF_RESULT_IMAGE_TOO_LARGE error code as we previously discussed. But I found that it would take a lot of plumbing (changing three or four functions) to surface the new error code from here. (The grid and item sizes are easy to handle.) In the end I decided to just reply on diagnostic messages and not add a new error code.